hedera-node/docs/design/app/workflows.md

```diff
@@ -86,4 +86,114 @@ The query workflow consists of the following steps:
+Examples include the `Configuration`, `RecordListBuilder` and `TokenContext`.
+Dagger provides all the objects that can be constructed in this scope [here](https://github.com/hashgraph/hedera-services/tree/develop/hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/flow/txn/modules) 
+and [UserTxnComponent](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/flow/txn/UserTransactionComponent.java)
+takes all the inputs that are needed to execute the user transaction.
+- **UserDispatchScope** - Objects that are created once for each user transaction that will be dispatched. 
+Examples include the `SingleTransactionRecordBuilder` for user transaction and `FeeContext`.
+Dagger provides all the objects that can be constructed in this scope in [UserDispatchModule](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/flow/dispatch/user/modules/UserDispatchModule.java) and `UserDispatchComponent`.
+and [UserDispatchComponent](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/flow/dispatch/user/UserDispatchComponent.java)
+takes all the inputs that are needed to create the user dispatch.
+- **ChildDispatchScope** - Objects that are created once for each child transaction dispatch. 
+Examples include the `ReadableStoreFactory` and `ChildFeeContext`. 
+Dagger provides all the objects that can be constructed in the [ChildDispatchModule](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/flow/dispatch/child/modules/ChildDispatchModule.java) 
+and [ChildDispatchComponent](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/flow/dispatch/child/ChildDispatchComponent.java)
+takes all the inputs that are needed to create the child dispatch.
+
+
+#### HandleWorkflow overview:
+
+![Basic Overview](images/handle_basic_overview.png)
+
+The `HandleWorkflow` class is responsible for handling the platform transaction and providing the record stream.
+The overall high level steps are as follows:
+
+1. Calls `BlockRecordManager` to update the new consensus time for the user transaction, puts the lastBlockInfo in state if needed.
+when blocks
+2. `UserTxnWorkflow` is called to handle the transaction and provide record stream
+3. Externalizes the record stream items
+4. Update metrics for the handled user transaction
+
+
+#### UserTxnWorkflow overview:
+
+![User Transaction Workflow](images/user_txn_workflow.png)
+
+1. **SkipHandleWorkflow**: If the transaction is from older software, the transaction will be skipped handling by calling `SkipHandleWorkflow`. 
+This writes a record with `BUSY` status and adds to record cache
```

Comment:
Is `BUSY` correct for obsolete transactions?
That would be very confusing to me if I were the client developer.  Something like `TRANSACTION_EXPIRED` would make a lot more sense to me.

Reply:
The transaction is not handled. Its not same as `TRANSACTION_EXPIRED`. We can add a new response code. But mono-service was adding `BUSY` status, which is almost same as it is being throttled and not handled.

Reply:
If the transaction is submitted by using older software version. This happened as an edge case during one of the upgrades and we added this check. Sure, will discuss what response will be better with team.